include/opt-sched/Scheduler/sched_basic_data.h

```diff
@@ -405,15 +404,25 @@ class SchedInstruction : public GraphNode {
   bool FindDef(Register *reg) const;
   // Returns whether this instruction uses the specified register.
   bool FindUse(Register *reg) const;
-  // Retrieves the list of registers defined by this node. The array is put
-  // into defs and the number of elements is returned.
-  int16_t GetDefs(Register **&defs);
-  // Retrieves the list of registers used by this node. The array is put
-  // into uses and the number of elements is returned.
-  int16_t GetUses(Register **&uses);
-
-  int16_t GetDefCnt() { return defCnt_; }
-  int16_t GetUseCnt() { return useCnt_; }
+
+  int16_t NumDefs() { return defCnt_; }
+  int16_t NumUses() { return useCnt_; }
+
+  llvm::ArrayRef<Register *> GetDefs() {
+    return {defs_, static_cast<size_t>(defCnt_)};
```

Comment:
This has the same issue as before right? It just extends it to other parts of the code. I thought we could make the underlying datastrcuture (defs_,uses_) a vector. Is this a lot of work to implement? I know it's in list from #47.

Otherwise I can just review the graphtrans portion of the change without this latest commit.

Reply:
I am planning to pull this commit out into its own PR; I just haven't gotten around to that yet.

What do you mean by "same issue as before"? This doesn't copy the underlying datastructure at all. It's just a pointer + size. This also allows the underlying datastructure to be converted into a `vector<Register *>` without any change for users. I did not want to implement conversion to a vector yet because it's tricky to ensure the change doesn't break anything.

Reply:
Okay this makes sense to me.